lib/web_scraper.dart

```diff
@@ -1,167 +1,220 @@
-            '$variableName *=.*?;(?=([^\"\']*\"[^\"\']*\")*[^\"\']*\$)',
-            multiLine: true);
-        //  Iterate all matches
-        Iterable matches = re.allMatches(script.text);
-        matches.forEach((match) {
-          if (match != null) {
-            // list for all the occurence of the variable name
-            var temp = result[variableName];
-            if (result[variableName] == null) {
-              temp = [];
-            }
-            temp.add(script.text.substring(match.start, match.end));
-            result[variableName] = temp;
-          }
-        });
-      }
-    }
-
-    // Returning final result i.e. Map of variable names with the list of their occurences.
-    return result;
-  }
-
-  /// Returns webpage's html in string format.
-  String getPageContent() => _response != null
-      ? _response.body.toString()
-      : throw WebScraperException(
-          'ERROR: Webpage need to be loaded first, try calling loadWebPage');
-
-  /// Returns List of elements found at specified address.
-  /// Example address: "div.item > a.title" where item and title are class names of div and a tag respectively.
-  List<Map<String, dynamic>> getElement(String address, List<String> attribs) {
-    // Attribs are the list of attributes required to extract from the html tag(s) ex. ['href', 'title'].
-    if (_response == null) {
-      throw WebScraperException(
-          'getElement cannot be called before loadWebPage');
-    }
-    // Using html parser and query selector to get a list of particular element.
-    var document = parse(_response.body);
-    var elements = document.querySelectorAll(address);
-    // ignore: omit_local_variable_types
-    List<Map<String, dynamic>> elementData = [];
-
-    for (var element in elements) {
-      var attribData = <String, dynamic>{};
-      for (var attrib in attribs) {
-        attribData[attrib] = element.attributes[attrib];
-      }
-      elementData.add({
-        'title': element.text,
-        'attributes': attribData,
-      });
-    }
-    return elementData;
-  }
-}
-
-/// WebScraperException throws exception with specified message.
-class WebScraperException implements Exception {
-  var _message;
-  WebScraperException(String message) {
-    _message = message;
-  }
-  String errorMessage() {
-    return _message;
-  }
-}
+/*
+  Developed by Tushar Ojha
+  GitHub: https://github.com/tusharojha
+  Twitter: https://twitter.com/tusharojha
+  Feel free to improve the web_scraper library.
+*/
+
+library web_scraper;
+
+import 'dart:async';
+
+import 'package:html/parser.dart'; // Contains HTML parsers to generate a Document object.
+import 'package:http/http.dart'; // Contains a client for making API calls.
+
+import 'validation.dart';
+
+/// WebScraper Main Class.
+class WebScraper {
+  // Response Object of web scrapping the website.
+  var _response;
+
+  // Parsed document from the response inside the try/catch of the loadWebPage() method.
+  var document;

```

Comment:
If we are saving the `document` in a separate variable to make the parsing efficient for the package. Then, we don't need to carry the `_response` variable as all the assertions can be updated accordingly to use the `document` variable. 

Also, make the `document` variable private by adding _ in prefix as we have already had a function to get the document code i.e. `getPageContent`.

Reply:
Don't you still need the response so you can check if the user used the page loading method before using any of the other methods? Not changing this until you let me know, aight?

Reply:
Resolving this as I've replaced the response variable with the document variable. Think i just didn't understand what you meant last time and checking the code today, I just did that and after reading here I understood what you previously requested.